Sources/SwifterSwift/UIKit/UIButtonExtensions.swift

```diff
@@ -167,34 +167,72 @@ public extension UIButton {
-            imageEdgeInsets = UIEdgeInsets(top: 0, left: -insetAmount, bottom: 0, right: insetAmount)
-            titleEdgeInsets = UIEdgeInsets(top: 0, left: insetAmount, bottom: 0, right: -insetAmount)
-            contentEdgeInsets = UIEdgeInsets(top: 0, left: insetAmount, bottom: 0, right: insetAmount)
-        }
+    func setImageLeftOfText(spacing: CGFloat) {
+        let insetAmount = spacing / 2
+        imageEdgeInsets = UIEdgeInsets(top: 0, left: -insetAmount, bottom: 0, right: insetAmount)
+        titleEdgeInsets = UIEdgeInsets(top: 0, left: insetAmount, bottom: 0, right: -insetAmount)
+    }
+
+    /// SwifterSwift: Set the image to the right of the text
+    ///
+    /// - Parameters:
+    ///   - spacing: spacing between title text and image.
+    func setImageRightOfText(spacing: CGFloat) {
+        // https://stackoverflow.com/questions/2451223/#7199529
+        guard
+            let imageSize = imageView?.image?.size,
+            let text = titleLabel?.text,
+            let font = titleLabel?.font else { return }
+
+        let titleSize = text.size(withAttributes: [.font: font])
+
+        let titleOffset = -(imageSize.width + spacing / 2)
+        titleEdgeInsets = UIEdgeInsets(top: 0, left: titleOffset, bottom: 0, right: -titleOffset)
+
+        let imageOffset = -(titleSize.width + spacing / 2)
+        imageEdgeInsets = UIEdgeInsets(top: 0, left: -imageOffset, bottom: 0, right: imageOffset)
+    }
+
+    /// SwifterSwift: Set the image above the text
+    ///
+    /// - Parameters:
+    ///   - spacing: spacing between title text and image.
+    func setImageAboveText(spacing: CGFloat) {
+        guard
+            let imageSize = imageView?.image?.size,
+            let text = titleLabel?.text,
+            let font = titleLabel?.font else { return }
+
+        let titleSize = text.size(withAttributes: [.font: font])
+
+        let titleOffset = -(imageSize.height + spacing)
+        titleEdgeInsets = UIEdgeInsets(top: 0.0, left: -imageSize.width, bottom: titleOffset, right: 0.0)
+
+        let imageOffset = -(titleSize.height + spacing)
+        imageEdgeInsets = UIEdgeInsets(top: imageOffset, left: 0.0, bottom: 0.0, right: -titleSize.width)
+    }
+
+    /// SwifterSwift: Set the image below the text
+    ///
+    /// - Parameters:
+    ///   - spacing: spacing between title text and image.
+    func setImageBelowText(spacing: CGFloat) {
+        guard
+            let imageSize = imageView?.image?.size,
+            let text = titleLabel?.text,
+            let font = titleLabel?.font else { return }
+
+        let titleSize = text.size(withAttributes: [.font: font])
+
+        let titleOffset = -(imageSize.height + spacing)
+        titleEdgeInsets = UIEdgeInsets(top: titleOffset, left: -imageSize.width, bottom: 0.0, right: 0.0)
+
+        let imageOffset = -(titleSize.height + spacing)
+        imageEdgeInsets = UIEdgeInsets(top: 0.0, left: 0.0, bottom: imageOffset, right: -titleSize.width)
     }
```

Comment:
```suggestion
    func setImageBelowText(spacing: CGFloat) {
        guard
            let imageSize = imageView?.image?.size,
            let text = titleLabel?.text,
            let font = titleLabel?.font else { return }

        let titleSize = text.size(withAttributes: [.font: font])

        let titleOffset = -(imageSize.height + spacing)
        titleEdgeInsets = UIEdgeInsets(top: titleOffset, left: -imageSize.width, bottom: 0.0, right: 0.0)

        let imageOffset = -(titleSize.height + spacing)
        imageEdgeInsets = UIEdgeInsets(top: 0.0, left: 0.0, bottom: imageOffset, right: -titleSize.width)

        let horizontalInsets = min(imageSize.width, titleSize.width) / -2
        let verticalInsets = (min(imageSize.height, titleSize.height) + spacing) / 2
        contentEdgeInsets = UIEdgeInsets(top: verticalInsets, left: horizontalInsets, bottom: verticalInsets, right: horizontalInsets)

        sizeToFit()
    }
```

Reply:
normal:
<img width="275" alt="normal" src="https://user-images.githubusercontent.com/12928482/98517973-b36cc700-22a9-11eb-983d-cce0e9394295.png">

your:
<img width="278" alt="your" src="https://user-images.githubusercontent.com/12928482/98518038-ca131e00-22a9-11eb-8fde-243e0c34c989.png">

Before you:
<img width="283" alt="Before you" src="https://user-images.githubusercontent.com/12928482/98518181-034b8e00-22aa-11eb-8bac-787f2d799e57.png">

I think users need to adjust the size of the button.

Reply:
This is with `sizeToFit`:
![Screen Shot 2020-11-09 at 23 43 13](https://user-images.githubusercontent.com/924764/98605575-f2b90900-22e5-11eb-92dc-79c642d48c0a.png)

And this is without `sizeToFit`:
![Screen Shot 2020-11-09 at 23 43 37](https://user-images.githubusercontent.com/924764/98605597-f9478080-22e5-11eb-84ac-874431c62133.png)

Does the first one really make more sense to you?

Reply:
Normally you'll be using auto-layout on buttons, so it actually won't make a difference whether or not you call `sizeToFit`. If for some very bizarre reason you're manually sizing everything I can understand why `sizeToFit` would break your frames. In that case you can instead call `sizeToFit` in the unit tests (so that the tests can pass), but then please add in the code comments (documentation) that manually-sized buttons will need to call `sizeToFit`.

Reply:
There is a gap between testing in the playground and testing in actual projects, you can see my description below.

Reply:
It's fine you can remove `sizeToFit`.